### ISSUE TITLE
Misleading License Name & Clarifications

### ISSUE THREAD

**Issue by LokiChaos** [jakimfett/MMPLv2#28]
Regardless of the intended goals of the license (as I have seen discussed in prior issues, creating a permissive license is not your intent):  it clearly does not adhere to the spirit of the MMPLv1 by adding restrictions, specifically on redistribution.  Calling it the MMPLv2 misrepresents the nature of the license.  

Also the section "The User grants the Author all rights to any contribution."  Is this a copyright assignment clause?  Or simply that any contibutions are implicitly licensed perpetually to the author?  The fact that it later mentions a CLA implies the latter to be the case.   This should be explicitly clear.  

The section on derivates also is unclear, should "The Derivative complies with the Mod license." be taken to mean that the derivativs must also be licensed under the same terms?  Or does the mean something else?

**Comment by jakimfett** [comment]
Thanks for voicing your concerns, @LokiChaos. Let me see if I can address some of them.

Speaking to your first point:  
My understanding of the [original MMPL (v1.0.1)](http://www.mod-buildcraft.com/MMPL-1.0.txt) is that it intends to promote openness of code sources, ease of distribution, and end user accessibility. The MMPLv2 does have several restrictions, but they are targeted at abuses and loopholes in the MMPLv1. 
The main difference (if I'm reading v1.0.1 correctly) is in the restriction on commercial distribution. This change is one that the original authors of the MMPLv1 might not agree with, but I'm more than willing to discuss it with them if they feel that preventing an uninvolved 3rd party from making a profit off mod author's work is antithetical to their aims.

For your second point:
Yes, that is indeed intended as copyright assignment. It's one of the areas that I intend to discuss with my lawyer when I have a chance, as I know there's specific rules about assigning copyright. 
As for the intent of this section, it's to give the author more control over the code itself, and (hopefully) to prevent any repeat of the wolverness debacle.
The CLA portion later in the same section does not imply either of these...simply that if the author so chooses, they can specify a CLA assigning some rights back to whoever wrote the code.

Third point:
Thanks for catching this. The intent is that derived works need to be licensed under the same license. Would you like to clarify it, and submit a pull request?

**Comment by LokiChaos** [comment]
The intent of the MMPLv1, as I understand it is to promote and protect the [four freedoms](https://en.wikipedia.org/wiki/Free_software#Definition) to the extent that is possible while creating a work intrinsically linked to Minecraft.  It seeks to protect those rights for the mod author, contributers, and users.  The MMPLv2 essentially denies and/or restricts those rights for the two latter groups.  

Attempting to restrict monetized distribution makes it incompatible with many OSI-approved licenses.  This limits the code's ability to interact with other more permissive licenses because it discriminates on the basis of purpose.  It's more than "closing a loophole"  While I morally oppose the exploitation of others works for profit (or even fame/credit), the task of crafting a permissive license that sufficiently limits such activities is a non-trivial task (this is the sort of thing you should leave to a very good contract law lawer).  I also don't view the phenomenon of these offensive sites as a licensing issue, but a larger community problem.

Copyright assignment needs to be EXPLICITLY clear.  What happened with Bukkit is a pretty unique case and a result of a failure to properly handle licensing from the start.  The FSF even answers in it's [FAQ](https://www.gnu.org/licenses/gpl-faq.html) how to prevent these exact sorts of problems (GPL linking exceptions).  Concerns were repeatedly raised years before the DMCA notice and they willfully ignored them and took no action to rectify it.  There are ways to handle similar issues that do not involve handing over the contributor's copyrights to the mod author.  I categorically oppose such an assignment clause for several reasons:

A) It may not be possible for the contributer to hand over the rights.  eg: Their contribution is a derivative of another parties work (say under a very permissive license such as MIT or WTFPL).  Nothing forbids one from incorporating the code, but the contributer lacks any legal standing to transfer the rights: they are not her's to hand over.  Similarly if the mod author includes code from others, they can possibly relicense it under other terms (depending on the terms of the original license), but cannot claim ownership of copyrights to the code others wrote.
B) Handing over the copyrights of the contributions over to the author gives them rights that a license generally does not grant.  Specifically the absolute right to relicense:  An author ALWAYS has the option to license their code under different terms, or even concurrently with other licenses.  If a contributer makes a submission under the MMLPv2, and the copyrights are assigned to the mod author, nothing forbids the mod author from then distributing that contribution under any set of terms and is free to cease distribution under the terms of the MMLPv2.  They could even perform a monetized distribution themselves and/or cease sharing the source and any contributers have forfeited any standing to object.  Copyright law does not restrict the rights holder in any way: no matter what license you write as copyright law only applies to other parties.
C) On assigning rights to the mod author the contributer no longer has the right to distribute their contribution under any terms except those set/approved by the mod author.  For example while the MMPLv2 give them the right to "re-use" their code elsewhere, however they have handed the exclusive right to set the license to the mod author, they can no longer do so without seeking approval from the mod author.   This effectively forbids them from contributing the code to other projects (even completely unrelated ones) or using it in a project of their own.  Similarly, they have lost the standing to object should the other projects terms are violated.

You are free to attempt to craft a license with whatever terms you wish in an attempt to solve what you perceive as a problem stemming from a deficiency in the existing licensing choices.  I just strongly feel that calling such a license "MMPLvX" disingenuous, confusing, and misleading.

**Comment by cuchaz** [comment]
I can't say I disagree with anything at all in that last post. If not by reserving rights, how would you suggest the mod repost problem be solved?

**Comment by LokiChaos** [comment]
I believe in trying to fix the underlying causes that encourages/promotes the behavior, a proative solution.  At some point you have to accept that mods will be reposted, regardless of the license.  At best the license affords you the option to attempt to seek redress but it is fundamentally reactive and at some point the costs of enforcement exceeds what many mod creators can afford.

I view the main problem is the community has failed to adequately provide a central, trusted source for mods.  Curse is sorta stepping up and some launchers have attempted to address this, but there has not been a complete solution and these solutions in some ways have come too late.  I somewhat view this as a failure on Mojang's part to not provide, or at least condone some framework for mod distribution.  Looking at other communities, specifically ones with permissive licenses:  Malware-ridden, outdated, paywalled, and ad-covered mirrors of many FOSS projects exist.  However, it is generally a minor issue as the FOSS community has created the proper infrastructure to distribute software reliably and safely: a package manager.  Due to permissive licenses distros can handle hosting their own copies, and it is understood they hold some responsibility for their safety and up-to-date-ness.  The very fact that going to a search engine and typing in "minecraft mods" or "minecraft [modname]" is often the first step many players take to locating mods is a direct contributer to creating the opportunity for these offensive sites to exist.  The fact a prospective mod user ever goes to a generic search engine represents a failure.  If MC launchers offered URIs to community-approved mod hosting services, or better, a package-manager to handle automated installation of modpacks and/or individual mods (this would also make modpacks better as they could be reduced to a list of mods+versions and far smaller archive of config/* and you don't have pointless blob archives of all the mod jars for a pack), the vast majority of the low hanging fruit would be denied to the ad-sites and paywalled ones.  They are in it for the money, and if you take steps to limit/reduce it's profitability they will move onto other things.  

As for malware sites, Google and other entities provide means to report such sites.  A license is a poor solution to these since they have abandoned any pretense of legality anyway. 

Plain old plagarism is likely such a small subset that you could viably enforce it reactively. Howver, most permissive licenses do still require attribution so they still permit you to seek redress.

Finally, the above isses are compounded by the fact that "offical" mod downloads can appear via ad-fly, dropbox, or megaupload links is a problem in itself.  It only makes it harder for an uninformed party to determine the safety of a hosted mod.

**Comment by cuchaz** [comment]
That's a very useful analysis. Thanks for taking the time to write it out.

I've often toyed with the idea of creating some kind of package manager for mods that was distributed and run by mod authors themselves rather than third parties (eg Curse). I think Asie has given this topic some thought too. Even if I (or we) did build such a thing, why would people use it? Getting Minecraft players to actually use such an endorsed platform seems tricky. I guess that's where the #StopModReposts movement can help out.

Indeed, the search results for Minecraft mods most often often lead to these ripoff websites. Are you saying mod authors just need to chamption a single mod package manger/distribution platform and then do SEO to get better search rankings? That might not be a bad idea at all.

**Comment by cuchaz** [comment]
Ah, actually, I missed one crucial point. Ideally, users wouldn't search for mods on google. We'd have a mod "app store" of sorts built into the package manager. Hmm... this will require some more thought. =)

**Comment by LokiChaos** [comment]
It doesn't have to be a /single/ solution, but I know MultiMC is working on a system called [quickmods](http://multimc.org/posts/introducing-quickmods.html) and Curse is working on a system themselves.  I do belive that any such a system should be open and egalitarian by design. 

While I belive that just a direct wget is superior than forcing the user to engage in click-though, the latter does accomodate ad-fly and or pickier modders' terms.  Licensing of mods has also contributed to the problem, it makes it harder to create a centeralized listing + tools like these and not violate the sometimes quirky and odd terms.  This is one reason I am so pedantic about software licensing:  A poor license choice hurts everyone (even the mod creator!) and trying to roll your own just creates more corner cases the community must deal with.

**Comment by cuchaz** [comment]
I like the quickmod solution better since it at least allows distributed file hosting. It also has an elegant simplicity to it. The unfortunate downside is that I would imagine the clickthrough process makes the user eperience a little bit annoying. It would be better I think if some kind of monetization scheme could be built right into the package manager, and then the proceeds distributed to any author who wants their fair share. With completely open accounting, of course. Public books for everyone!

Also, the more energy I spend on trying to write this license, the more I realize how hard it is, and how I don't really have the skills to do it properly. I've always wished there was a better way, and maybe you just showed me what it might be. =)

**Comment by LokiChaos** [comment]
I would be wary of some intrinsic monetization, again license choices by MC modders create issues.  I could see potential objections to CC-NC mods being listed in such a system, even if they are not directly being sold or used to promote ads. 

Software licensing is a /very/ tricky beast.  I'm not even an expert, just someone who has read a lot over the years.  Contract law and Copyright law are entire legal specialties, and software licensing requires both in addition to a sound technical understanding.  

As to the larger issue of shady mod reposts.  It should be attacked like any other problem: identify and address the underlying circumstances that cause it.  This approach is far more work, requires the community to cooperate, and takes time, but it is ultimately the only effective solution.  Just trying to ban it, or craft licenses that forbid it doesn't change why it happens and a large portion of those engaging in it simply do not care whether or not it's permitted/legal and are operating from beyond where the wronged parties can take action.

**Comment by cuchaz** [comment]
Creative commons licenses are generally not useful for software projects. I'm not aware of many projects that release software under that license, so hopefully that won't be a problem?

All the same, I'd prefer some kind of opt-in scheme where mod authors put their mods in the "monetization zone" in the package manager only if they want to. That way, by default, everything stays non-monetized and the NC people can still be happy.

I don't know if a "monetization zone" distinction in the package manger would be a sufficient enough barrier for the NC people to claim NC though. I guess it all depends on what the mod authors want.

**Comment by LokiChaos** [comment]
I can think of about 15 mods that are CC in some form off the top of my head.  ProfMobius' mods are all CC-SA-NA 4.0 for example, though I doubt he would raise any objections.  CC is awful for software, but it gets used often enough in MC modding.

**Comment by jakimfett** [comment]
You've made a lot of good points, @LokiChaos. I need to take some time (which, due to IRL events I don't have at the moment) and read through carefully and make sure I actually understand all of what you're saying.

That being said, at this point, I'd be most interested in @SpaceToad weighing in on this, as he actually wrote v1, or possibly @SirSengir (dunno if he helped write it or not), or anyone else involved in the Buildcraft project when the MMPLv1 was written. Essentially, it's all well and good for us to express opinions about the original MMPL, but they're exactly that...opinions. Only the original author(s) know the real intent behind the license, and whether or not they're fine with this license being called the MMPLv2 may be a deciding factor in whether or not I choose to rename it.

**Comment by cuchaz** [comment]
I'll stop hijacking this thread now, but I've tried to spin off the mod package manager discussion into the #StopModReposts thread on MFC. Feel free to contribute there:
http://www.minecraftforum.net/forums/mapping-and-modding/minecraft-mods/mods-discussion/2314237-list-of-sites-stealing-minecraft-content-updated?comment=784

**Comment by bdew** [comment]
I agree that the name is extremely misleading, this license is not open source and has nothing to do with the original MMPL.

**Comment by Strikingwolf** [comment]
It was inspired by the MMPL. And it is semi-open source. We need a better way to express it. One of the main goals is to make the license as permissible as possible while putting mod authors on legal ground to DMCA people like 9minecraft and the ilk.

**Comment by bdew** [comment]
> It was inspired by the MMPL

Yet it goes against the spirit of the MMPL and open source. 

Also calling something "Xv2" is very different than "Inspired by X" - that's why it's misleading.

> semi-open source

For some values of semi. It contradicts at the very least part 1 (Free Redistribution) and 3 (Derived Works) of the [Open Source Definition](http://opensource.org/osd-annotated)

**Comment by Strikingwolf** [comment]
@bdew That is the problem. We can't make a true open-source license that fulfills the main mission of this license. We want this to be as permissible as possible, but we don't want to give up some of that legal ground

**Comment by bdew** [comment]
That's your prerogative, but in that case you shouldn't name it MMPLv2 which would be misleading to both mod authors and users.

**Comment by Strikingwolf** [comment]
@bdew if you look at some of the other issues we have been thinking about changing the name and putting inspired by the MMPL.

**Comment by projectilemotion** [comment]
It's just occurred to me how terrible this license is. It restricts the rights people have *anywhere* else in the open source/free software world, and further it encourages people not to give users the rights that the FSF and other organizations have been fighting for more than 20 years to give people - the four freedoms.

I can't imagine why a mod author, aside from being consumed by such large ego that he or she may feel the need to keep all these rights to him or herself, would use such a license, especially as it's non-standard (there are probably more than 100 other Minecraft-specific licenses), non-legalese (there appears to be no involvement of legal consultancy) and most importantly *not FLOSS*.

**Comment by Strikingwolf** [comment]
@projectilemotion there are actually many reasons, so addressing your points.

* Of course it is non-standard, it's new and not even really fully finished
* The legalese/legal consultancy will happen after it is constructed and I believe we will keep a "simple" version for people who want to decide if they will use the license
* It is _not FLOSS_ but it provides a fundamental thing that mod authors want, which is the ability to take down websites redistributing and making money from it and misleading users, while trying to remain as open as possible. It's trying to be some sort of middle ground

As for reasons to use it
* It does a good job of catching loopholes to problems of distribution by sites such as 9minecraft
* It is fairly open for it's mission
* It allows people to gain all the benefits of OS and extend these benefits to their users without putting their users safety and their livelihood at risk. While OS can avoid the latter, it takes more time to gain a userbase that truly enjoys the mod if a quarter of them are using sites like 9minecraft

As for me even if/when this gets finished I will be using MIT because I don't need that restriction as I don't want to make money or to get "credited" for making mods, but I understand why people would

**Comment by coolsquid** [comment]
@projectilemotion It seems to me like this license might not be intended for you. That does not, however, mean that it is a bad license. If you want want your freedoms, I suggest you make use of the MIT license or another applicable license. And remember, that for the right holders, not using a license and thereby not granting any rights is far easier than finding and applying an applicable license. Be happy about the rights you get, nobody has to give you any.

**Comment by jakimfett** [comment]
@projectilemotion the purpose of this license is to provide a legally enforceable license tailored to the unique needs of Minecraft modders. I've been extremely careful with the wording, I've consulted legal professionals, and I've ensured that I'm taking into account everything I know about modding and the Minecraft community.

As stated by @Strikingwolf and @coolsquid, nothing is forcing **anyone** to use this license. If you don't like it...or if your needs are different...go use something else, be free, don't concern yourself with the restrictions of this custom license.

However...if you're looking for a license that gives content authors the ability to smack down abuses (eg mod rehosting or monetization by 3rd parties) while still giving the community itself (the players, modpack makers, and modding enthusiasts) near total freedom...you might want to take another look at it.

The restrictions this license puts on the end user are pretty dang simple. In essence, don't make money off the mod author's work without permission, and don't be a jerk. While it's less permissive than a license like MIT, it's less permissive for thoroughly researched and extensively discussed reasons. Reasons like "users getting malware from repackaged mods" and "redistribution sites hosting old outdated buggy versions (which reflects poorly on the mod and results in useless bug reports and outdated complaints)".

As I've stated elsewhere, I respectfully disagree with the OSI definition of "Open Source". That said, this license is an open source (note the lower case usage for a non-proper noun) license in the most literal sense...eg, "the original source code is made freely available and may be redistributed and modified". The restrictions of this license are intended to deal with the compiled binary, not the source code.

As for your statements about ego...they are outside the scope of a discussion of the merits of one technical license vs another.

I would, however, be interested in seeing a list of the other Minecraft specific licenses...I'm aware of less than twenty, and I'm astounded that somehow I've missed over 80% of all the licenses relating to this topic which I've put so much effort into researching.

**Comment by RazzleberryFox** [comment]
Decocraft is not open source at all, not until it is finished, when it is finished I think I will definitely use this license as people have already tried stealing my models and decompiling my code illegally, as well as reposting my mod and trying to sell items in game for real money. This is a great license!

**Comment by iyra** [comment]
@RazzleberryFox

> as people have already tried stealing my models and decompiling my code illegally

This is in fact not a justification for using a restrictive license like this. This is not sound reasoning. Can you explain the logic of this? How does people "stealing" your models lead to using a restrictive license such as this? How does people decompiling your code illegally lead to using a license such as this?

If you don't give people the right to use (or whatever you mean by "steal") your models at a low cost (i.e only the requirement of share-alike and/or attribution), people will _still_ do it. You noted that people do it illegal - so what's to stop them from doing it illegally this time too? Simply because it has  yet another "don't be a jerk", non-legalese license slapped onto it?

> as well as reposting my mod

Thousands of other people have no issue with their mods or software being reposted. In fact, within the free/open source software world, sharing is actively encouraged. It's a shame to see someone want to stifle this, especially as there's been a shift with projects other than Minecraft adopting more permissive licenses, which take all restrictions of distribution aside from attribution.

> sell items in game for real money

Can you clarify this please? Unless you want to control how people run your code (which is unclear to me), why does what people sell matter? And why "real money" rather than tokens of some sort?

I'm just as against people "stealing" things as you probably are, but why do you think this license will stop it? At least with the GPL, BSD etc. people know what the license does because the license is either popular or understandable to humans (while remaining legalese). Further, to take away the right to decompile code is a truly horrible thing to do; it goes into the same territory Microsoft and Apple stay in with their EULAs. The enforceability of such a clause is even more dubious, especially in non-US countries.

At least give the users and Minecraft modding community the respect you would like to get back. If you don't want people to do bad things with your code, be nice to people and it'll pay off. The lack of respect this license shows towards users, while at the same time proclaiming "don't be a jerk" (an incredibly vague term, probably designed to shoot down anyone who disagrees with the license creators if the person is "being a jerk" by some arbitrary standard) and saying that if you don't agree to this "don't be a jerk" you should "GTFO" (this it seems is a classic "jerk" thing to do) is horrible.

Do yourself, your sanity, the Minecraft modding community and your heart a favor.

**Comment by jakimfett** [comment]
> ...not a justification for using a restrictive license like this...

Actually, "doing things that the license would allow me to crack down on" is pretty much textbook perfect reason to use aforementioned license.

> ...How does people "stealing"...How does people decompiling...lead to using a license such as this?

Because this license explicitly calls out Minecraft modding related abusive behaviors (in a way that, for example, Creative Commons does not) and allows a content author to take action against them.

> ...If you don't give people the right to use your models at a low cost...people will still do it

If I'm not mistaken, content creators are allowed to do whatever they want with their work, including but not limited to "only use it for my own purposes" and "require other people to pay arbitrary amounts of money before using my work".

> ...(or whatever you mean by "steal")...

Making statements like this towards a well known and respected content creator is not unacceptable, and if you do so in the future, I'll be removing the corresponding section of your comment. **Keep your interactions civil or I will take administrative action.**

> ...what's to stop them from doing it illegally this time too...

The license you use is directly related to what action you can take against people violating a license. This isn't rocket surgery. Someone violates the license, you take action against it.

> ...yet another "don't be a jerk"...

In the context of a license, the words you use are important. You're confusing this license with my other license, the [Don't Be a Jerk non-commercial license](https://github.com/jakimfett/dbaj).

> ...non-legalese license slapped onto it..

If by "non-legalese" you mean "not legally enforceable", you're dead wrong. Every license is legally enforceable. The license may make it difficult or impossible to violate the terms (eg MIT or WTFPL), but if a user violates the terms of the license, you can then take steps to enforce it.  
If by "non-legalese" you meant "understandable by individuals who aren't legal professionals", then you're absolutely right. This license is intended to be straightforward and to address known issues of abuse in the modding community, while still being readable and understandable by younger or less knowledgable end users.

> Thousands of other people have no issue with their mods or software being reposted...

Please provide a source for this. Given the popularity of the "[Stop Mod Reposts](http://www.minecraftforum.net/forums/mapping-and-modding/minecraft-mods/mods-discussion/2314237)" movement, I respectfully disagree that "thousands of people" are actually ok with it, especially when taken in the context of the Minecraft modding community, which has possibly a thousand or two _total_ mod authors.

> ...within the free/open source software world, sharing is actively encouraged. It's a shame to see someone want to stifle this...

Please re-read the license. It explicitly encourages people to use and share the mod, and the only "stifling" it does is identifying specific abuses and attempting to eliminate them. If you feel that the license prohibits legitimate activities, please provide specifics of things that are being prohibited (in a new issue report) and I will take steps to address the issue.

> ...why does what people sell matter...

Because we as content creators have the right to say "no, you **cannot** make a profit from the hours of effort I put into creating this thing". Because when someone makes money from content they didn't create without permission, it's sending a massive "screw you" to the individual who created the content. Because we as content creators are tired of seeing our content used to make money for people without acknowledgement of our work.  
On a personal note, I've literally never turned down anyone who asked if they could use some of my modding content to make money. In the case of MinecraftEdu and others, I've actively worked with them to make sure the version of my content they're using is the absolute best for their purposes. They, in turn, provide visibility and acknowledgement that I put in effort to create the content. That's how a legitimate business relationship between content creators and personal or business entities who wish to use their content is _supposed_ to work.

> ...why do you think this license will stop it...

Stop it? Unlikely. Make some people think twice, and give us legal grounds to require the stolen content be removed? Already happening, see previous comments about the Stop Mod Reposts movement.

> ...to take away the right to decompile code is a truly horrible thing to do...

Which is precisely why this license explicitly **allows** decompiling, see section 4 of the license, "[Right to Examine](https://github.com/jakimfett/MMPLv2/blob/master/LICENSE.md#4-right-to-examine). If you're going to provide a critique of this license, at least have the decency to read and understand it first.

> ...enforceability of such a clause...

If you can remove it from legitimate mod hosting sites (dropbox, onedrive, CurseForge, etc) and remove it from google search results, you've effectively removed it from the internet for the average internet user. As Dropbox, Microsoft, Google, and Curse are USA-based companies, this license allows quite effective action to be taken to remove infringing content.

> ...give the users and Minecraft modding community the respect you would like to get back...

Yup, basically this. I'm glad we agree on such a foundational principle of life. Mutual respect. It's the normally-unwritten contract between individuals that says "if you don't screw me over, I won't screw you over". This license is, in essence, calling out places where someone has said "I stole your content, screw you, what'cha gonna do?", and providing a legal standing for replying "making sure that people quickly forget you".

> The lack of respect this license shows towards users...

Please provide specific cases where an end user is disrespected by such things as "do not screw over content creators" and "please look through my source code, just get my permission before you reuse it".

> ...while at the same time proclaiming "don't be a jerk"...

You're confusing my [DBaJ](https://github.com/jakimfett/dbaj) license with this one again.

> ...designed to shoot down anyone who disagrees with the license creators...

No. The DBaJ license (which, I'll note is **not** this license) is designed to allow a content creator to define what they consider jerk-ish behavior, and act accordingly. Yes, this allows a content creator to be a jerk about enforcing the license, but it also gives people like me to ignore harmless jerks (or people with an overdeveloped sense of sarcasm) and only enforce against the people who are actively screwing us over...but that's a conversation to have [over in the DBaJ issues](https://github.com/jakimfett/dbaj/issues) section, **not** here. 

> Do yourself, your sanity, the Minecraft modding community and your heart a favor.

Actually, I did. I did myself a favour by creating a license that addresses specific abuses of a system. I did my sanity a favour by having actual legal professionals look over the license and provide feedback to make it legally sound. I did the modding community a favour by participating in the "Stop Mod Reposts" movement and making all my mod source code available on github. 

I'm doing my heart a favour by systematically addressing your entire statement, and finishing with this bit:  
If you're so passionate about making it better...where's your pull request resolving the specific criticisms you've had for this license?

**Comment by RazzleberryFox** [comment]
@projectilemotion 

The only thing I am going to say to you (because I have a ton of work) is to clarify the stealing... people have sold my models in game for in game currency that gets bought with real life currency on servers. THIS is not acceptable. And it also comes in direct violation of the MINECRAFT EULA. Also others have decompiled the model code, and used a converter to reproduce my models in Blender format, and then attempted to sell "packs" with my models in them for people using cinema 4D. These packs contained 20 of my models each and were sold for $5 USD. I was only able to get my models taken down because I had a closed source mod. Now. I am willing to make it open source, but I am asking for respect. Deal with it.

**Comment by iyra** [comment]
@jakimfett

> Actually, "doing things that the license would allow me to crack down on" is pretty much textbook perfect reason to use aforementioned license.

It's not textbook perfect at all. You choose a license to give people liberties, not to try and stop people from doing things they are already doing. Now that people are doing something with _one_ license, what makes you think they'll stop doing it if you change the license to something else? Especially when they realize you changed the license to try and stop them rather than adapt to the conditions.

> If I'm not mistaken, content creators are allowed to do whatever they want with their work, including but not limited to "only use it for my own purposes" and "require other people to pay arbitrary amounts of money before using my work".

Of course they're allowed to do whatever they want. The problem is where other people don't agree with that, and they breach the terms of the license. If you make the license amicable, people are more likely to obey it. I'm willing to bet the number of GPL/BSD license infringements is way less than the number of people who knowingly violate Microsoft's EULA on Windows, even adjusted for population size.

> Making statements like this towards a well known and respected content creator is not unacceptable

Like what? I asked the question of what he/she means by "steal". It's very unclear, especially as "steal" has a nuanced meaning in law. Further, would it be any more acceptable for me to use this statement against a "well known and respected" person instead of, say, an "unknown and unrespected" person such as me?

I intend to keep my comments civil. Sorry you felt that it was otherwise.

> Someone violates the license, you take action against it.

Against some random anonymous guy on the Internet who has nothing to lose because he dosen't associate his on-line identity with his real life? What if it's done via proxy on 4chan or something? This whole idea of enforcing Minecraft mod licenses is out of proportion. They seem to exist only to tell people what's acceptable and what's not. If we're serious, no Minecraft modder is going to hire lawyers and come after someone who violated their license, unless it's a gross violation. Even then, probably telling the guy to stop would be more effective.

Bad people don't obey licenses, and most likely know how to protect themselves. It's not rocket surgery. This is even if you know for sure such a license would hold up in court.

> Every license is legally enforceable.

Enforcable, but still up to interpretation. This is what I mean. If it has room for interpretation, any lawyer worth his salt will pick up on it. Compare your license with something like the GPL or even the BSD license. Minecraft licenses are notorious for being written by people who know nothing about how licenses are interpreted.

> Stop Mod Reposts movement

This is laughable and ultimately seems to be something made by people new to the idea of free software/open source. The way to stop people from distributing mods with malware isn't to stop people from distributing it at all. It's to use trademarks and educating Minecraft users that you should only download mods from reputable sources. Thanks to education, the idea of downloading things from Download.com. The whole "stolen money" thing is what the MPAA etc. base their "predictions" on before they sue people for sharing Game of Thrones or whatever.

Also, here's a list of free mods (okay, not quite a thousand but I'm sure there are more, and more people who support the idea anyway):

Tinkers construct, Natura, EnderIO, Tinkers Construct, Blood magic, GardenStuff, Applied energistics, FastLeafDecay, randomstuff, CoFHLib, forestry, OpenComputers, BetterFps, NotEnoughItems, CodeChickenCore, CodeChickenLib,  StorageDrawers, Chisel, AsieLib, Computronics, catwalks, VersionChecker, SimplyJetpacks, WailaHarvestability, Waila (*I think), rftools, EnderIOAddons, rftools, mcjtylib, Flaxbeard'sSteamPower,  immibis-microblocks, immibis-core, redlogic, tubestuff, MapWriter, Extracells, Botania (Uses a almost foss with a non comertical clause license), Buildcraft and logistic pipes (uses a almost foss but requires you own a legal minecraft license), Magic bees, StacksOnStacks, AgriCraft, waterhooks, rogugelike, sprinklesforvanilla, Ancient Trees, Realistic-Terrain-Generation, TerraFirmaCraft, CartLivery?, Tropicraft I think, Ex Nihilo, EnderTech.

See how many FOSS mods there are that haven't had this "stealing" problem? See how many free software projects, literally the biggest in the computing world (GNU/Linux, OpenBSD etc.) that haven't had the stealing problem? Actually, OpenBSD has had the stealing problem - people have used the code but not given anything back except attribution. And you know what? **The OpenBSD guys say they wouldn't prefer it, but they accept it's the most free solution**. The GNU GPLv3 would eliminate this "stealing" problem that the OpenBSD guys had, if only for code rather than monetary "theft".

You could also just use a CC license that disallows commercial use. I wouldn't prefer it, but it does exist. Even Buddhist monks use it on their translated texts.

> prohibits legitimate activities

A legitimate activity is modifying any part of the code and releasing your modified version with (as the GPL enforces) or without (BSD, ISC, Unlicense etc.) corresponding source code, while giving credit to the original author, and not claiming that you made the original.

> Because we as content creators

Literally anyone can be a "content creator". There isn't some group you represent if you say "we". I don't think I have that right, and I'm a content creator.

> Because when someone makes money from content they didn't create without permission, it's sending a massive "screw you" to the individual who created the content. 

What's so bad about a "screw you"? It's the Internet. If you don't like something, fight back with better features, a more permissive license etc. I use the GPL on my software, but if people take it, I don't get so mad. I don't take it as a "screw you". I send an e-mail or letter and tell them about it.

> The Internet is not for sissies. -- Paul Vixie
> 
> Because we as content creators are tired of seeing our content

You're doing it again.. I'm not tired of it. In fact, I encourage people to make money so long as they re-distribute the source. This means that I benefit (I can merge changes back in) and they benefit (a little money on the side). Why should I stop someone else's benefit? Lecturers put their notes online, but so that they can be seen to people regardless of the university they go to. Every user of the GPL and BSD licenses is fine with this. Hey, even Linus Torvalds is.

> I've literally never turned down anyone who asked if they could use some of my modding content to make money.

Though I don't think they ought to need to ask, I'm pleased.

> That's how a legitimate business relationship between content creators and personal or business entities who wish to use their content is supposed to work.

That's similar to how the GPL works too.

> Make some people think twice

Exactly!

> and give us legal grounds to require the stolen content be removed?

Only on places that will take things down if you say you're the creator.

> Which is precisely why this license explicitly allows decompiling, see section 4 of the license

I was replying to RazzleberryFox, sorry. I wasn't referring specifically to this license when I said that. He seemed not to like the idea of decompilation (which is really just studying the program). I don't think allowing decompilation ought to be necessary.

> Mutual respect.

Like releasing source code of any modifications so that anyone can see them? Giving others the right to distribute their cool modifications?

> "if you don't screw me over, I won't screw you over"

The GPL's mentality is "if you screw me over, so long as you obey the license, nobody actually gets screwed over".

> "please look through my source code, just get my permission before you reuse it"

This. I don't think that's respectful, especially as the standard of respect expected in the FOSS world is much higher.

> If you're so passionate about making it better

I'm not a lawyer, and I'm not good at writing licenses. People with more experience, such as yourself (and I say this non-sarcastically) can take up and make a decision as to implement the recommendations or not. It's no skin of my nose if you don't; I just want to see a more free Minecraft modding community.

@RazzleberryFox

> THIS is not acceptable.

Why isn't it? Even the GNU GPL allows this.

> And it also comes in direct violation of the MINECRAFT EULA.

That's Mojang's job to worry about; not yours.

> These packs contained 20 of my models each and were sold for $5 USD.

Did they have attribution? If they have attribution, why is it a problem?

> but I am asking for respect

And the rest expect respect from you - by using a nice license. Respect is not a one way thing, I'm afraid.

> Deal with it.

Uh, okay.

Addendum: The "if I can't do <x>, then I won't let anybody do <x>!" is a really bad way to behave in a community. Just because a mod developer dosen't know how to market something (and gives it away for free), why does this mean nobody should profit off it? If someone knows how to profit from something, he's adding his own skill to the process.

**Comment by jakimfett** [comment]
> You choose a license to give people liberties

No. **You** may choose licenses this way, but that is not how I (and many other) individuals choose licenses.
The purpose of this license is to provide a relatively open legally enforceable license for Minecraft modders that helps close loopholes in other licenses. To put it bluntly, this license is intended to stop people from monetizing content they don't own.

> ...what makes you think they'll stop doing it if you change the license to something else?

Other than the fact that _it's working_? Based on efforts I've taken surrounding my non-monetization license clauses, 3rd parties no longer show up on the 1st page of google results for "Minechem". That's a major win, as less than 6% of google users actually visit the 2nd page.

> ...adapt to the conditions...

Abuse and misuse of unlicensed content is not something content creators should need to adapt to. "Adapt to the wrong being done to you" is never a good option.

> ...other people don't agree with that, and they breach the terms of the license...

Uhuh. And as I've stated previously, enforcing a license does work. Especially when the license is crafted to meet the needs of the individuals who want to play the game, or make modpacks, or...yaknow, actually be a community. Literally the only people I'm excluding are the people making commercial misuse of content.

> If you make the license amicable, people are more likely to obey it.

Source, please.

> Like what? I asked the question of what he/she means by "steal". It's very unclear, especially as "steal" has a nuanced meaning...

Using the word "steal" was possibly less precise than it could be. Context is a wonderful thing, however. Steal has a nuanced meaning, yes, but what I was calling you out on was the "whatever you mean by" bit. You basically said "nobody is actually stealing anything, your complaint is invalid", and that's what I was identifying as out of line. When a content creator brings their voice to a conversation on why addressing content theft and abuse of unlicensed content, the **only** appropriate response is support. Doing otherwise is a slap in the face to them, in that they spoke up about their creations being abused to pay other people's bills (illegally, I might add), and you responded with a "haha, right, whatever" attitude.

> ...would it be any more acceptable for me to use this statement...

No. No it would not. Twisting my words like this is not appreciated. I said nothing about you as a person, or your relative visibility (or lack thereof) in the community.

> Against some random anonymous guy on the Internet...

No, because it's not the random anonymous that is the problem. It's a website. Pick a website, any website. Chances are, they're hosted in the USA or the domain registrar is based in the USA. Which means if the website owner (which you can always contact via DNS records, one way or the other) doesn't respond, you go up the chain...to their host, if their host doesn't respond, you contact the domain name registry, if that doesn't work, you work through the process of getting the domain registrar in trouble with the legal system. At some point (normally long before you get to the domain registrar level) you find someone who cares more about the profit they're making from _all the other customers_ more than the customer violating the content license.

Literally the only problem is identification...and because the sites spend so much time trying to get their google search rankings boosted, it's like shooting fish in a barrel. Do a google search of a mod name, click through the first three or so pages, send a half dozen "You have 14 days to remove my content from your sites", and call it a day.

> What if it's done via proxy...

Again...if it's not showing up on google search results, it probably doesn't matter, and might as well not exist from the average user's perspective. Sharing a random file on 4chan, or via reddit, or whatever, isn't monetization. This license (which I've explained multiple times to you now) exists to stop the abuses of content for illegal monetization.

> ...telling the guy to stop would be more effective...

Yup, which is why the first step in enforcing a license against an abusive site is always "send a carefully worded email specifying the license and the infringing content". It's not a one-and-done thing...it's a process.

> ...still up to interpretation...any lawyer worth his salt...Minecraft licenses are notorious for being written by people who know nothing about how licenses are interpreted

Which is why, as I've emphasized before multiple times, I have my lawyer provide input on the changes to the license. I've compared this license to many, many others. I didn't just one day decide "hey, I'm gonna write a license". I've spent a couple of years studying law, and I back up my understanding with an actual licensed legal professional. Interpretation only gets you so far in a court room, and the statements in this license are straight forward, to the point, and there's not a lot of wiggle room. The people abusing content know that. For them, it's cheaper to just take something down than risk going to court.

> This is laughable and ultimately seems to be something made by people new to the idea of free software/open source.

The Stop Mod Reposts movement isn't about free software. It's about content creators upset that people are abusing their content. It's right there in the name, the movement is about stopping a specific abuse, that of mod reposting. I don't know how exactly you got such a wrong idea about the goals and intention, but I'm reasonably sure it's not because of the name.

> ...use trademarks and educating Minecraft users...

Oh, so you do support the Stop Mod Reposts movement! That's exactly what it's trying to do...educate the end user on what's bad, and give content authors easily accessible tools for shutting down abuses.

> ...here's a list of free mods...not quite a thousand but I'm sure there are more...

Erm. You're going off on a tangent. I didn't say "there aren't freely available mods", I said that I need a source that "thousands of other people have no issue with their mods or software being reposted". Available for free (via legitimate sources) doesn't mean a mod author is fine with their content being monetized by a 3rd party. There are indeed some mod authors who don't care, but that's not "thousands of other people". It's one here, two there, six there, four there...hardly the thousands you claimed. So again I say...please provide a source.

> how many free software projects...haven't had the stealing problem...

Yeah, not gonna go there. I've seen both ways work for different projects, and I'm not going to turn this into a debate about free vs non-free, as that's completely outside the scope of this license. Nothing in this license has anything to do about "freely available", it's about preventing specific subsets of abuse and being as nonrestrictive as possible in all other areas.

> A legitimate activity is modifying any part of the code and releasing your modified version...

Which [you're allowed to do](https://github.com/jakimfett/MMPLv2/blob/master/LICENSE.md#7-right-to-derive) as long as you're not monetizing it, can you please just re-read the license, I'm getting _really_ sick of having to point out where the bits you're ranting about are actually allowed under this license. RTFM or GTFO. Simple as that.

> ...anyone can be a "content creator"...

Yup, and in this case, I'm representing a fairly large group of content creators who have said "this license protects us in ways no other license does, and we're choosing to use it". In the context of what I've been saying, it's the group of Minecraft modders who are tired of having our content abused. If you're note part of that group...feel free to not use the license.

> What's so bad about a "screw you"?

Because communities are built on mutual respect. Because screwing people over is morally reprehensible. Because being bad to people is bad.

> It's the Internet. If you don't like something, fight back with better features, a more permissive license etc.

Or, we could fight back with a license that smacks down the people abusing our content, and actually win the battle for a bit.

> I use the...I don't get...I don't take it as...

I'm happy that you've found a license that works for you. Kindly stop proselytizing your chosen license in a thread devoted to improving a completely different license. If you don't like the license...don't use it. If you have nothing constructive to contribute, kindly refrain from being actively destructive.

> You're doing it again.. I'm not tired of it...

I've not made any claims that I represent you. I do represent a group of individuals who are tired of this sort of content abuse.

> Why should I stop someone else's benefit?

Because it's actively to our detriment, for reasons long and extensive, the highlights of which are outdated or incorrect information, malware, creating invalid bug reports, and weakening the ability of a mod creator to update their mod and fix problems affecting the community.

> Only on places that will take things down...

Which, as I've previously established, is in the USA, the EU, and anywhere else with decent content copyright laws. Given how much of the internet's architecture is based in these places, and how many of the end users reside in one of these places, a license like this allows a content creator to pretty much wipe abusive content off the face of the internet.

> The GPL's mentality...

Which, as I've stated elsewhere, has zero to do with this license's mentality.

> I'm not a lawyer, and I'm not good at writing licenses...

I have a lawyer, and I am good at writing legal documentation, licenses among them. I've personally spent tens of dozens of hours fine tuning the wording of this license to ensure that the community, as a whole, wins. I've spent hundreds of hours on social media, IM, and IRC communicating with content creators, end users, and uninvolved 3rd parties to figure out what will give the most back to the community.

Given your repeated insistence that this license isn't free, isn't open, and is both unenforceable and overly restrictive, please re-read the license and then provide **only** constructive criticism consistent with the license's stated intentions and goals in the future.

**Comment by jakimfett** [comment]
Addendum: I'm responding on Razz's behalf.

> Why isn't it? Even the GNU GPL...

As we've previously established, this license isn't the GPL. The reason it's not acceptable is because a content creator has said "I do not authorize my content to be used in this manner", and because it's their content, they get to say what is ok and what isn't.

> That's Mojang's job to worry about...

Except we as modders have agreed to the EULA, part of which is that no part of the game will be monetized individually (there's a [long](https://mojang.com/2014/06/lets-talk-server-monetisation/) [FAQ](https://mojang.com/2014/06/lets-talk-server-monetisation-the-follow-up-qa/) on the Mojang site, go read it), and as such we can report EULA violations and take steps to smack down violations that involve our content.

> And the rest expect respect from you - by using a nice license.

As licenses go, this is one of the nicer ones. Please re-read it, then compare the actual things you're allowed to do (essentially everything but monetize content that isn't your own and be actively malicious) against something like "All Rights Reserved".

> Respect is not a one way thing...

Very true. And "I shall take your content, monetize it, and give you nothing" is not respect. As 3rd parties have opted to do things like this which screw us over, we take actions like specifically call them out in our licenses.

> The "if I can't do , then I won't let anybody do !" is a really bad way to behave in a community.

Yeah, no, that's not a valid criticism. Razz isn't, and didn't, say "because I can't do it, nobody else can either". What was actually said is that content was created, abused, and then future content suffered because of it.

> Just because a mod developer dosen't know how to market something...why does this mean nobody should profit off it?

I'm sorry, I fail to see how "3rd party abuses content to make money, then uses money gained from those abuses to abuse more" has anything to do with marketing. Content creators deserve to be compensated for their work. 

Content creators can decide where, how, and when their content is used. If someone profits from a content creator's work without permission and against their wishes, that "someone" is a bad person. Emphasis on permission. If I say "no, don't use this", and you do it anyway, you're a bad person.

**Comment by iyra** [comment]
> Available for free (via legitimate sources) doesn't mean a mod author is fine with their content being monetized by a 3rd party.

The licenses which the mods I named use have no restriction on monetising the code, aside from the ones that don't allow commercial use. This leaves some situations open:
- the mod author doesn't care if people make money from it
- the mod author is happy for people to make money from it
- the mod author hasn't considered people making money from it
- the mod author is only using that license because he's forked it from someone else

I'd say that if the author is releasing it for free in the first place, then he will most likely fit into the first and second category, or begrudgingly the fourth category.

> It's one here, two there, six there, four there...hardly the thousands you claimed. So again I say...please provide a source.

I'll give you that "thousands" may be an exaggeration. But at the same time, my point was that there's a lot of mod authors who would be fine with people selling their modifications, or apathetic. If they aren't one of these two, they would have specified a non-commercial clause or something like that.

> Content creators deserve to be compensated for their work. 

Then they should make people pay for the mod, or receive compensation in the form of source code from anyone who modifies and re-distributes the mod.

I think we have differences in our way of thinking. Your license, though I heavily disagree with the principles, has an aim that I see may suit some people. And I've realised that's fine. The best I can do is encourage people not to use it, and I'll do that in other places.

I'm sorry for using your time; my only hope was to encourage the concerned people to make it a more "open" license in a way that I would think the community would benefit more from.

Thanks for taking the time to reply, and truly have a nice day, @jakimfett.

**Comment by jakimfett** [comment]
> ...aside from the ones that don't allow commercial use...

Ah. You realize that monetization = commercialization, right? A non-commercial license is a non-monetization license, and vice-versa.

> ...if the author is releasing it for free in the first place...

I release my mods for free. So do many people I know. Many of the people I know aren't in any of the four categories you listed.

> ...my point was that there's a lot of mod authors who would be fine with people selling their modifications, or apathetic...

Based on my interactions with the modding community, this is incorrect. Based on the thousands of people who have downloaded the browser plugins that warn you if you're visiting a 3rd party rehosting site, this is incorrect. Based on conversations with people at conventions, this is incorrect.

> Then they should...

**No**. Absolutely not. Content creators deserve to be compensated for their work. Full stop. If I, as a content creator, say "Do not monetize my content", the _only_ appropriate response is to refrain from monetizing my content. 

I release my mods because I want people to play them, enjoy them, combine them with other mods, and just generally have fun and do cool stuff with them. Restricting access to the free downloads available via official sources is a stupid as putting up a fence around a public park and charging admission because someone let their dog poop in it. The solution is to penalize the people abusing the free service, not punish all the people who treated it respectfully.

> ...best I can do is encourage people not to use it...

There's a line between encouraging people to use a given license because it's better, and actively dissing a license you dislike. I would encourage you to refrain from crossing that line, as it's one of the primary distinctions between "enthusiast for something" and "asshat who tears down other people".

> ...my only hope was to encourage the concerned people to make it a more "open" license...

You've yet to provide a single piece of actionable input, criticism, or even a suggestion for improvement. If you're actually trying to make the modding community a better place, stop tearing things (and people) down, and start actually building, contributing, and actively making them better.

**Comment by elifoster** [comment]
> Now that people are doing something with one license, what makes you think they'll stop doing it if you change the license to something else?

So we should just get rid of all laws and never make new ones, right?
We should never update or make changes to any sort of legally binding contract, guideline, or anything of the sort? Your logic is severely flawed.

**Comment by FayeAlephNil** [comment]
@elifoster is correct. The logic is wrong, completely. If you want to convince someone of something, make a better argument

**Comment by joepie91** [comment]
> As I've stated elsewhere, I respectfully disagree with the OSI definition of "Open Source".

There's nothing to disagree with. They practically coined the term. They decide what it means.

> That said, this license is an open source (note the lower case usage for a non-proper noun) license in the most literal sense...eg, "the original source code is made freely available and may be redistributed and modified".

That is extremely misleading. Don't call something open-source unless you're willing to provide the associated guarantees and benefits.

**Comment by coolsquid** [comment]
> There's nothing to disagree with. They practically coined the term. They decide what it means.

Organizations don't decide what terms mean. The people using them do. I've met more people using the term in its literal meaning than I've met people using the OSI definition. Most people don't even know what OSI, nor its definition of open source, is.

**Comment by joepie91** [comment]
@coolsquid They created the term. Yes, they define what it means - for the same reason any trademark can't just be used arbitrarily (from an ethical point of view, not just a legal one). And this matters, because the term "open-source" is meant to convey that certain guarantees and benefits are provided, which is not the case for non-OSI-compliant licenses.

"People decide how they use language" only applies to regular words, not to "names" or "brands". If you allow that, language becomes meaningless. People _do_ have certain expectations for "open-source software", and those expectations are not met with a non-OSI-compliant license either.

I can't see any non-malicious reasons to try and redefine "open-source" either. The only purpose I can see for doing that is to gain the public approval and popularity that the name brings, without delivering on the goods. That is not okay.

**Comment by coolsquid** [comment]
"Open source" is very much a normal term, not a brand or name. It is frequently (I'd even say mostly) used without any connection to OSI.

> People do have certain expectations for "open-source software"

*some people

> I can't see any non-malicious reasons to try and redefine "open-source" either

Nobody is redefining it. "software for which the original source code is made freely available and may be redistributed and modified" would be the normal definition in both common speak and [your average dictionary](http://www.oxforddictionaries.com/definition/english/open-source). The fact that OSI made the term is _completely_ irrelevant to its definition.
